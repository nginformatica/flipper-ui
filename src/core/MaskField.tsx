import React, { Component } from 'react'
import TextField, { TextFieldProps } from './TextField'
import NumberFormat from 'react-number-format'

interface MaskFieldProps {
    mask?: string
    type?: 'text' | 'tel' | 'password'
    decimalSeparator?: string
    format?: string
    decimalScale?: number
    thousandSeparator?: boolean | string
    fixedDecimalScale?: boolean
    customInput?: React.ComponentType<TextFieldProps>
}

class MaskField extends Component<TextFieldProps & MaskFieldProps> {
    public render() {
        const { customInput, ...otherProps } = this.props

        return (
            // Although react-number-format allow use of additional props,
            // shows problem with some props like have been do this
            // actually on flipper-ui. (e.g. errors treatment)
            <NumberFormat
                { ...otherProps }
                customInput={ customInput || TextField }>
            </NumberFormat>
        )
    }
}

export default MaskField
