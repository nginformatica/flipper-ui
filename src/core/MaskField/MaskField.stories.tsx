import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import InputAdornment from '../InputAdornment'
import MaskField from '.'

export default {
    title: 'MaskField',
    component: MaskField,
    argTypes: {
        helperText: { control: 'text' },
        value: { control: 'text' || 'number' }
    }
} as ComponentMeta<typeof MaskField>

const Template: ComponentStory<typeof MaskField> = args => (
    <MaskField {...args} />
)

export const Default = () => <MaskField placeholder='Description' />

export const MaskFieldWithInputAdornment = Template.bind({})
MaskFieldWithInputAdornment.args = {
    fixedDecimalScale: true,
    decimalScale: 2,
    label: 'Price',
    decimalSeparator: ',',
    thousandSeparator: '.',
    InputProps: {
        startAdornment: <InputAdornment position='start'>$</InputAdornment>
    },
    InputLabelProps: { shrink: true },
    children: <MaskField />
}
