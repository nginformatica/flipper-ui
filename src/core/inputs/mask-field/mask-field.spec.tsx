import React, { act } from 'react'
import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import MaskField from '.'

describe('MaskField', () => {
    it('should render with default TextField without customInput', async () => {
        render(<MaskField placeholder='Description' />)

        const input = screen.getByPlaceholderText(
            'Description'
        ) as HTMLInputElement

        await act(async () => await userEvent.type(input, '123'))

        expect(input.value).toBe('123')
    })

    it('should match snapshot', () => {
        const { container } = render(<MaskField placeholder='Description' />)

        expect(container).toMatchSnapshot()
    })
})
