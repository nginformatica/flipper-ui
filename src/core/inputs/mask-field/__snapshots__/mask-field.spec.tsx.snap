// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskField should match snapshot 1`] = `
<div>
  <div
    class="sc-dmyCSP kvMDoa"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
      inputmode="numeric"
      title=""
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        data-testid="text-field"
      >
        <input
          aria-invalid="false"
          autocomplete="off"
          class="MuiInputBase-input MuiOutlinedInput-input makeStyles-outlinedInput-17"
          placeholder="Description"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-23"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
